website/docs/r/nsxt_distributed_firewall_rule.html.markdown

```diff
@@ -0,0 +1,121 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_distributed_firewall_rule"
+sidebar_current: "docs-vcd-resource-nsxt-distributed-firewall-rule"
+description: |-
+  The Distributed Firewall rule allows user to segment organization network entities by creating firewall rules.
+---
+
+# vcd\_nsxt\_distributed\_firewall\_rule
+
+The Distributed Firewall rule allows user to segment organization network entities by creating
+firewall rules.
+
+Multiple rules defined with this resource will **not be created in parallel** because Cloud Director
+API provides no direct endpoint to create a single rule. To overcome this,
+`vcd_nsxt_distributed_firewall_rule` calls an "update all rules" API endpoint for each single rule,
+and this call is serialized to avoid data conflicts.
+
+!> There is a different resource
+[`vcd_nsxt_distributed_firewall`](/providers/vmware/vcd/latest/docs/resources/nsxt_distributed_firewall)
+that can manage all firewall rules in one resource. One should use **only one of**
+`vcd_nsxt_distributed_firewall` or `vcd_nsxt_distributed_firewall_rule` as using both will result in
+unexpected firewall configuration.
+
+## Example Usage
+
+```hcl
+data "vcd_vdc_group" "existing" {
+  org  = "my-org" # Optional, can be inherited from Provider configuration
+  name = "main-vdc-group"
+}
+
+data "vcd_nsxt_app_port_profile" "FTP" {
+  context_id = data.vcd_nsxt_manager.main.id
+  name       = "FTP"
+  scope      = "SYSTEM"
+}
+
+resource "vcd_nsxt_distributed_firewall_rule" "r1" {
+  org          = "my-org"
+  vdc_group_id = vcd_vdc_group.test1.id
+
+  name        = "rule1"
+  action      = "ALLOW"
+  description = "description"
+
+  source_ids           = [vcd_nsxt_ip_set.set1.id, vcd_nsxt_ip_set.set2.id]
+  destination_ids      = [vcd_nsxt_security_group.g1-empty.id, vcd_nsxt_security_group.g2.id]
+  app_port_profile_ids = [data.vcd_nsxt_app_port_profile.FTP.id]
+}
+
+resource "vcd_nsxt_distributed_firewall_rule" "r2" {
+  org          = "{{.Org}}"
+  vdc_group_id = vcd_vdc_group.test1.id
```

Comment:
nit, but maybe you could also change this in the distributed_firewall docs?
```suggestion
  vdc_group_id = data.vcd_vdc_group.test1.id
```

Reply:
Thanks, fixed them.